src/Ansatz/Chain.jl

```diff
@@ -227,6 +227,30 @@ function Base.rand(rng::Random.AbstractRNG, sampler::ChainSampler, ::Type{Open},
     Chain(Operator(), Open(), arrays)
 end
 
+Tenet.contract(tn::Chain, query::Symbol, args...; kwargs...) =
+    Tenet.contract!(copy(tn)::Chain, query, args...; kwargs...)
+Tenet.contract!(tn::Chain, query::Symbol, args...; kwargs...) =
+    Tenet.contract!(boundary(tn), tn::Chain, Val(query), args...; kwargs...)
+
+function Tenet.contract!(::Open, tn::Chain, ::Val{:between}, site1::Site, site2::Site; direction::Symbol = :left)
```

Comment:
I don't see why you should separate implementations by boundary.

```suggestion
    Tenet.contract!(tn::Chain, Val(query), args...; kwargs...)

function Tenet.contract!(tn::Chain, ::Val{:between}, site1::Site, site2::Site; direction::Symbol = :left)
```

Reply:
Yep, that's true